llvm/lib/Transforms/Vectorize/VPlanPatternMatch.h

```diff
@@ -173,8 +173,8 @@ struct Recipe_match {
     if ((!matchRecipeAndOpcode<RecipeTys>(R) && ...))
       return false;
 
-    assert(R->getNumOperands() == std::tuple_size<Ops_t>::value &&
-           "recipe with matched opcode the expected number of operands");
+    if (R->getNumOperands() != std::tuple_size<Ops_t>::value)
+      return false;
```

Comment:
This change is due to m_GetElementPtr asserting. It's actually m_PtrAdd, but we can take care of it in follow-ups.